ytypes/list.go

```diff
@@ -424,6 +348,61 @@ func unmarshalList(schema *yang.Entry, parent interface{}, value interface{}) er
 	return nil
 }
 
+// makeKeyForInsert returns a key for inserting a struct newVal into the parent,
+// which must be a map. 
+func makeKeyForInsert(schema *yang.Entry, parentMap interface{}, newVal reflect.Value) (reflect.Value, error) {
```

Comment:
There seems to be enough logic here that we might want to test this function separately. Consider adding coverage (I'm OK for this to be a follow up if you feel it's best?)

Reply:
https://github.com/openconfig/ygot/issues/65
I think the coverage esp in list could use a revisit since the tests are out of date and several did not get fully fleshed out.